demisto_sdk/commands/common/docker_helper.py

```diff
@@ -467,22 +478,45 @@ def _get_python_version_from_env(env: List[str]) -> Version:
 
 
 @functools.lru_cache
-def get_python_version(image: Optional[str]) -> Version:
-    log_prompt = f"Get python version from image {image}"
-    logger.debug(f"{log_prompt} - Start")
-    if not image or "pwsh" in image or "powershell" in image:
+def get_python_version(
+    image: Optional[str], use_only_api: bool = False
+) -> Optional[Version]:
+    """
+    Get the python version of a docker image if exist.
+
+    Args:
+        image (str): the docker image
+        use_only_api (bool): whether to use only the dockerhub api when querying the python version if not found before
+
+    Returns:
+        Version: Python version X.Y (3.7, 3.6, ..)
+    """
+    logger.debug(f"Get python version from image {image} - Start")
+
+    if not image:
         # When no docker_image is specified, we use the default python version which is Python 2.7.18
         logger.debug(
             f"No docker image specified or a powershell image, using default python version: {DEFAULT_PYTHON2_VERSION}"
         )
         return Version(DEFAULT_PYTHON2_VERSION)
-    if match := PYTHON_IMAGE_REGEX.match(image):
-        return Version(match.group("python_version"))
+    if "pwsh" in image or "powershell" in image:
+        logger.debug(
+            f"The {image=} is a powershell image, does not have python version"
+        )
+        return None
+
+    if python_version := _get_python_version_by_regex(image):
+        return python_version
+
+    if use_only_api:
+        return _get_python_version_from_dockerhub_api(image)
+
     try:
+        logger.debug(f"Getting python version from {image=} from client")
```

Comment:
what's "from client"? which client are you referring to?

Reply:
changed it to be more understandable